tasks/rtloader.py

```diff
@@ -9,6 +9,25 @@ def get_rtloader_path():
     here = os.path.abspath(os.path.dirname(__file__))
     return os.path.join(here, '..', 'rtloader')
 
+def clear_cmake_cache(rtloader_path, settings):
+    """
+    CMake is not regenerated when we change an option. This function detect the
+    current cmake settings and remove the cache if they have change to retrigger
+    a cmake build.
+    """
+    cmake_cache = os.path.join(rtloader_path, "CMakeCache.txt")
+    if not os.path.exists(cmake_cache):
+        return
+
+    with open(cmake_cache) as cache:
+        for line in cache.readlines():
+            for key, value in settings.items():
+                if line.strip() == key + "=" + value:
+                    settings.pop(key)
```

Comment:
I think you should make a copy of settings, otherwise you are modifying it here.

Reply:
Indeed !

Reply:
Can you do the `.copy()` inside the function? I don't think the person calling the function should be responsible of doing the copy (or even knowing they have to do a copy).

Apart from that, looks good to me.